src/lib/Telemetry/telemetry.cpp

```diff
@@ -66,9 +63,18 @@ bool Telemetry::GetNextPayload(uint8_t* nextPayloadSize, uint8_t **payloadData)
     return false;
 }
 
+void Telemetry::UnlockCurrentPayload()
+{
+    if (payloadTypes[currentPayloadIndex].locked)
+    {
+        payloadTypes[currentPayloadIndex].locked = false;
+        payloadTypes[currentPayloadIndex].updated = false;
+    }
+}
+
 uint8_t* Telemetry::GetCurrentPayload()
```

Comment:
Can't we get rid of `GetCurrentPayload()`? Nothing else uses it and `UnlockCurrentPayload` already checks the lock. It doesn't do anything and just adds confusion. Just Unlock and move on in rx_main and eliminate GetCurrentPayload altogether. The pseudocode looks better I think
```
if (!Active)
    Unlock
    if (GetNext(&x))
        Sender.SetNext(x)
```
EDIT oops well I should have clicked a line at the end of the function so you could see it here, sorry about that.